pkg/serverless/startchecker/api_key_env_rule.go

```diff
@@ -0,0 +1,30 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package startchecker
+
+import "os"
+
+// ApiKeyEnvRule is used to ensure that an API KEY has been set via env variables
+type ApiKeyEnvRule struct{}
+
+const (
+	kmsAPIKeyEnvVar            = "DD_KMS_API_KEY"
+	secretsManagerAPIKeyEnvVar = "DD_API_KEY_SECRET_ARN"
+	apiKeyEnvVar               = "DD_API_KEY"
+)
+
+func (r *ApiKeyEnvRule) ok() bool {
+	if len(os.Getenv(kmsAPIKeyEnvVar)) == 0 &&
+		len(os.Getenv(secretsManagerAPIKeyEnvVar)) == 0 &&
+		len(os.Getenv(apiKeyEnvVar)) == 0 {
+		return false
+	}
+	return true
+}
```

Comment:
Not sure if this makes sense, but all of our environment variables eventually get loaded and we check `config.Datadog.IsSet("api_key")`. Would it make more sense to check for the presence of that, instead of checking the environment variables directly? What if they specify it via config?

Reply:
good point! I'll update